Add Configurable Global Error Handler to Spec

MitchellDumovic opened the pull request:
Fixes #

## Changes

Please provide a brief description of the changes here.

Related issues #

Related [oteps](https://github.com/open-telemetry/oteps) #

arminru committed:
Add resource semantic conventions for operating systems (#693)

reyang committed:
Fix typo (#702)

* fix typo

* rephrase

arminru committed:
Adapt semantic conventions for the span name of messaging systems (#690)

* Adapt messaging semantic conventions to include the operation in the span name

* Specify that the operation name must match the respective attribute if the suggested span name format is used

* Update span names in examples

* Consolidate definition of operation names into a separate section

* Consolidate capitalization

* Reference Conversation ID definition

* Consolidate MD syntax

* Organize definitions

* Organize definitions

* Reference definitions

* Allow artificial destinations as span name if neither destination name nor conversation ID are suitable

* Do not use conversation IDs for span name

* Wording

Co-authored-by: Christian Neumüller <christian.neumueller@dynatrace.com>

* Fix typo

* Add more links and guidance on messaging.operation

* Fix changelog

Co-authored-by: Christian Neumüller <christian.neumueller@dynatrace.com>

MitchellDumovic committed:
Add Configurable Global Error Handler to Spec

MitchellDumovic commented on specification/error-handling.md:
```diff
@@ -52,7 +52,50 @@ Another example may be a metric exposed by a `SpanProcessor` that describes the
+SDK authors SHOULD provide a mechanism to distinguish between different types of errors within a custom error handler. For example, 
+an end user should have the ability to selectively ignore errors surfaced by the `SpanExporter` while logging all other errors. 
+
+### Examples 
```
Not totally sure if this section is helpful or needed... it might not add a ton of value.

tedsuo replied:
I think this may be a case where Go is actually kind of confusing to non-gophers, but an example is always helpful.

kayousterhout replied:
Might be worth converting this to Java, now that you've implemented the Java one?

tedsuo commented on specification/error-handling.md:
```diff
@@ -52,7 +52,50 @@ Another example may be a metric exposed by a `SpanProcessor` that describes the
+
+The mechanism by which end users set or register a custom error handler should follow language-specific conventions. 
+SDK authors SHOULD provide a mechanism to distinguish between different types of errors within a custom error handler. For example, 
+an end user should have the ability to selectively ignore errors surfaced by the `SpanExporter` while logging all other errors. 
```
Error types are good, but I predict we will end up being asked to standardize the error types, or at least a common subset of them. I don't know if defining them now would be required or helpful for the PR to be accepted, but it will probably come up as groups start to implement this. If we don't really know what the types are yet, perhaps that should be noted; either in the spec or in the PR message.

carlosalberto replied:
I agree with Ted. I can imagine this exploding into a new sub topic. So let's go without the error types *for now*.

MitchellDumovic replied:
Is it still worth including this 'SHOULD' section here, or should I remove this altogether?

kayousterhout replied:
I'd probably be in favor of removing this because I'm not sure it adds much beyond what's already obvious / implied by the language

MrAlias committed:
Unify the batch span processor parameters (#708)

The parameter `maxExportBatchSize` refers to "export", but
`exporterTimeoutMillis` refers to the "exporter." This unifies on
"export" since they are both in reference to the same process.

jkwatson committed:
Add an addendum for allowing async/callback export return results. (#707)

dyladan committed:
specify how probability sampler is used with ParentOrElse (#704)

mwear committed:
Exception reporting specification (#697)

* Add semantic conventions for errors

* Update error.type description

Co-Authored-By: Armin Ruech <armin.ruech@gmail.com>

* update semantic conventions for exceptions

* add details for attributes; remove log.severity; add API details

* rename exceptions.md -> exception-reporting.md

* more accurately describe format for java

* alphabetize languages in stacktrace representation table

* prefer dynamic type over static type

* add log.severity convention

* Revert "add log.severity convention"

This reverts commit 137556e44bf74e6d76a17bd81943702a460cb9de.

* rename stacktrace -> exception.stacktrace

* move file to semantic conventions directory

* move RecordException to API

* fix formatting

Co-authored-by: Armin Ruech <armin.ruech@gmail.com>

kayousterhout commented on specification/error-handling.md:
````diff
@@ -52,7 +52,50 @@ Another example may be a metric exposed by a `SpanProcessor` that describes the
+```go
+// The basic Error Handler interface
+type ErrorHandler interface {
+	Handle(error)
````
super nit: might be nice to include a name here, for folks not familiar with Go (to make it obvious this is typed)

kayousterhout reviewed (approved):
this looks good to me to post - and I think having the example is helpful!!

carlosalberto committed:
Small clean up for Propagators. (#577)

yurishkuro committed:
Describe in-band and out-of-band terms (#714)

* Describe in-band and out-of-band terms

Signed-off-by: Yuri Shkuro <ys@uber.com>

* move anchors

Signed-off-by: Yuri Shkuro <ys@uber.com>

* Update specification/glossary.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Fix language

* Move to the top

Signed-off-by: Yuri Shkuro <ys@uber.com>

* Fix anchors

* Re-add anchor

Signed-off-by: Yuri Shkuro <ys@uber.com>

* fix link

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

MitchellDumovic committed:
addressed some review comments

bogdandrutu committed:
Clarify concurrency requirements for trace and metrics (#720)

* Clarify concurrency requirements for trace and metrics

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

* Update specification/concurrency.md

Co-authored-by: John Watson <jkwatson@gmail.com>

* Update specification/concurrency.md

Co-authored-by: Julian Tescher <julian@tescher.me>

* Update specification/concurrency.md

Co-authored-by: Yuri Shkuro <yurishkuro@users.noreply.github.com>

* Update concurrency.md

Co-authored-by: John Watson <jkwatson@gmail.com>
Co-authored-by: Julian Tescher <julian@tescher.me>
Co-authored-by: Yuri Shkuro <yurishkuro@users.noreply.github.com>

bogdandrutu committed:
Move concurrency requirements to the API files (#724)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Centralize attributes definition (#722)

* Centralize attributes definition

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

* Update specification/common/common.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Update specification/common/common.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Update specification/common/common.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Update specification/common/common.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Update specification/common/common.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Update specification/common/common.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

* Update specification/overview.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

tigrannajaryan committed:
Add log data model (#694)

* Add log data model

The content of data-model.md is primarily copied from OTEP 0097.
I removed sections that were only relevant in the context of OTEP
discussion and made minimal changes so that this is no longer
a "proposal" but a "specification".

* Address PR comments

bogdandrutu committed:
Split resource semantic conventions in smaller files (#727)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

AntonDzyk committed:
fix link 'Log Data Model' (#733)

bogdandrutu committed:
Add semantic conventions for all Kubernetes objects (#732)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

reyang committed:
DocFX sanity check (#742)

MitchellDumovic committed:
Change the default sampler to ParentOrElse(AlwaysOn) (#750)

* Change the default sampler to ParentOrElse(AlwaysOn)

* addressed some review comments

* more review feedback

Oberon00 committed:
Trace SDK: Provide definitions for readable and read/write span. (#669)

iNikem committed:
Remove warnings about Span.UpdateName (#754)

iNikem committed:
Close stale PRs (#744)

cijothomas committed:
Minor edits to the SpanContext (#741)

MitchellDumovic committed:
more post-sig updates with java example

MitchellDumovic committed:
Merge remote-tracking branch 'upstream/master' into error-handling-spec

MitchellDumovic committed:
Update specification/error-handling.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>

carlosalberto committed:
Update specification/error-handling.md

Co-authored-by: Christian Neumüller <christian+github@neumueller.me>
